cmd/podman/machine/init.go

```diff
@@ -79,18 +88,33 @@ func init() {
 	_ = initCmd.RegisterFlagCompletionFunc(IgnitionPathFlagName, completion.AutocompleteDefault)
 }
 
+func getSystemDefaultVMType() string {
+	if runtime.GOOS == "windows" {
+		return "wsl"
+	}
+
+	return "qemu"
+}
+
 // TODO should we allow for a users to append to the qemu cmdline?
 func initMachine(cmd *cobra.Command, args []string) error {
 	var (
 		vm     machine.VM
 		vmType string
 		err    error
 	)
+
+	vmType = getSystemDefaultVMType()
 	initOpts.Name = defaultMachineName
 	if len(args) > 0 {
 		initOpts.Name = args[0]
 	}
 	switch vmType {
+	case "wsl":
```

Comment:
Can we add another case for `qemu` and let the default case error?

Reply:
Same for the other commands.

Reply:
Sure I can add that

Reply:
Should this even be a `switch`? I’d expect a Go interface type with Qemu and WSL implementations would be cleaner and more performant — and more importantly significantly more maintainable, because adding a method to one implementation and forgetting about the other one (something that happens not quite infrequently with Linux-focused developers and testing) would be much easier to notice.

Reply:
I agree an interface would be better. I almost changed it but was originally trying to constrain the scope in this PR. It did need to expand quite a bit in the end anyway.

Reply:
done